src/cascadia/Remoting/Monarch.cpp

```diff
@@ -130,63 +139,239 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
+    //   _doHandleActivatePeasant, which will actually insert the
+    //   WindowActivatedArgs into the list we're using to track the most recent
+    //   peasants.
+    // Arguments:
+    // - args: the WindowActivatedArgs describing when and where the peasant was activated.
+    // Return Value:
+    // - <none>
     void Monarch::HandleActivatePeasant(const Remoting::WindowActivatedArgs& args)
     {
-        // TODO:projects/5 Use a heap/priority queue per-desktop to track which
-        // peasant was the most recent per-desktop. When we want to get the most
-        // recent of all desktops (WindowingBehavior::UseExisting), then use the
-        // most recent of all desktops.
-        const auto oldLastActiveTime = _lastActivatedTime.time_since_epoch().count();
-        const auto newLastActiveTime = args.ActivatedTime().time_since_epoch().count();
-
-        // For now, we'll just pay attention to whoever the most recent peasant
-        // was. We're not too worried about the mru peasant dying. Worst case -
-        // when the user executes a `wt -w 0`, we won't be able to find that
-        // peasant, and it'll open in a new window instead of the current one.
-        if (args.ActivatedTime() > _lastActivatedTime)
+        // Start by making a local copy of these args. It's easier for us if our
+        // tracking of these args is all in-proc. That way, the only thing that
+        // could fail due to the peasant dying is _this first copy_.
+        winrt::com_ptr<implementation::WindowActivatedArgs> localArgs{ nullptr };
+        try
+        {
+            localArgs = winrt::make_self<implementation::WindowActivatedArgs>(args.PeasantID(),
+                                                                              args.Hwnd(),
+                                                                              args.DesktopID(),
+                                                                              args.ActivatedTime());
+            // This method will actually do the hard work
+            _doHandleActivatePeasant(localArgs);
+        }
+        catch (...)
+        {
+            TraceLoggingWrite(g_hRemotingProvider,
+                              "Monarch_HandleActivatePeasant_Failed",
+                              TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+        }
+    }
+
+    // Method Description:
+    // - Helper for removing a peasant from the list of MRU peasants. We want to
+    //   do this both when the peasant dies, and also when the peasant is newly
+    //   activated (so that we don't leave an old entry for it in the list).
+    // Arguments:
+    // - peasantID: The ID of the peasant to remove from the MRU list
+    // Return Value:
+    // - <none>
+    void Monarch::_clearOldMruEntries(const uint64_t peasantID)
+    {
+        auto result = std::find_if(_mruPeasants.begin(),
+                                   _mruPeasants.end(),
+                                   [peasantID](auto other) {

```

Comment:
nit: `auto&&` will let the compiler decide what kind of parameter type is best. `auto` requires a copy (and the addref/release that you'd get by doing that)